bot/exts/info/code_snippets.py

```diff
@@ -0,0 +1,237 @@
+import logging
+import re
+import textwrap
+from urllib.parse import quote_plus
+
+from discord import Message
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.utils.messages import wait_for_deletion
+
+log = logging.getLogger(__name__)
+
+GITHUB_RE = re.compile(
+    r'https://github\.com/(?P<repo>.+?)/blob/(?P<path>.+/.+)'
+    r'#L(?P<start_line>\d+)([-~]L(?P<end_line>\d+))?\b'
+)
+
+GITHUB_GIST_RE = re.compile(
+    r'https://gist\.github\.com/([^/]+)/(?P<gist_id>[^\W_]+)/*'
+    r'(?P<revision>[^\W_]*)/*#file-(?P<file_path>.+?)'
+    r'-L(?P<start_line>\d+)([-~]L(?P<end_line>\d+))?\b'
+)
+
+GITLAB_RE = re.compile(
+    r'https://gitlab\.com/(?P<repo>.+?)/\-/blob/(?P<path>.+/.+)'
+    r'#L(?P<start_line>\d+)([-](?P<end_line>\d+))?\b'
+)
+
+BITBUCKET_RE = re.compile(
+    r'https://bitbucket\.org/(?P<repo>.+?)/src/(?P<ref>.+?)/'
+    r'(?P<file_path>.+?)#lines-(?P<start_line>\d+)(:(?P<end_line>\d+))?\b'
+)
+
+
+class CodeSnippets(Cog):
+    """
+    Cog that parses and sends code snippets to Discord.
+
+    Matches each message against a regex and prints the contents of all matched snippets.
+    """
+
+    async def _fetch_response(self, url: str, response_format: str, **kwargs) -> str:
+        """Makes http requests using aiohttp."""
+        try:
+            async with self.bot.http_session.get(url, **kwargs) as response:
+                if response_format == 'text':
+                    return await response.text()
+                elif response_format == 'json':
+                    return await response.json()
+        except Exception:
```

Comment:
You should avoid bare exceptions.

Reply:
Strictly speaking, a bare exception is a term attributed to `except:`. `except Exception:` is relatively much more justifiable. Anyway, I agree that the exception should be more specific in this case. In fact, the `get` should be using `raise_for_status=True` so that an exception is raised in the first place.